src/extractor/ConfigurationExtractor.ts

```diff
@@ -0,0 +1,113 @@
+import { countBy, toPairs, maxBy, capitalize } from 'lodash';
+import { ImplementationGuideStatus } from 'fsh-sushi/dist/fhirtypes/ImplementationGuide';
+import { Package } from '../processor';
+import { ExportableConfiguration, ExportableCaretValueRule, ExportableRule } from '../exportable';
+import { fshtypes } from 'fsh-sushi';
+
+export class ConfigurationExtractor {
+  static canonicalRegex = /^https?:\/\/(\w+)\.[\w.]+\/(.*)$/;
```

Comment:
I'm not sure we're agreed on how we name things like this, but should it be:
```ts
  static CANONICAL_REGEX = /^https?:\/\/(\w+)\.[\w.]+\/(.*)$/;
```
?

Reply:
Actually, you only use it in one place.  I think it would be cleaner just to inline it on line 19 than to use a constant.  That way when people read the code they don't need to bounce from line 19 to line 8 just to see what's going on.

Reply:
And... (sorry - my sort of stream-of-conscious review is made clear by my multiple comments)... A few things about the regex itself...

This would match `http://example.org/` but not `http://example.org`.  I think it should also match on the latter.

Because it only captures the first part of the domain, something like `http://www.example.org/our/ig` would result in `www.our.ig` -- but I would expect either `www.example.our.ig` or `example.our.ig` (if we wanted to put in special-case logic to ignore `www`).

Reply:
And to answer a question I _think_ you asked -- I believe that the general pattern you saw in mCODE will hold for any official HL7 IGs, but I'm not sure it will for non-HL7 IGs, which is why I want to be more flexible w/ the regex.

Reply:
Thanks for pointing out the issues with the regex. I had focused too much on the case of multipart top-level domains like `https://example.co.uk`, hence why was just taking the first part of the domain. Doing this well might require a list of TLDs to check against, but either way, I definitely need a lot more test cases for this function to make sure I'm giving something that is frequently correct and is always at least kind of close.

Reply:
I hadn't thought about `co.uk` and domains like that... Good point. That said, I think we need to recognize that we can't get it right 100% of the time -- and that it's not all that difficult for someone to fix it.  So don't go crazy trying to cover every edge case.  We probably just need a "good enough".